### PR TITLE
Delete for projects, organizations and data sources.

### PR DESCRIPTION
**Pull request by eanders**

### CODE REVIEW

File: app/models/grda_warehouse/data_source.rb [hunk 1]
```diff
@@ -372,6 +373,10 @@ def project_names
     projects.order(ProjectName: :asc).pluck(:ProjectName)
   end
 
+  def destroy_dependents!
+    organizations.map(&:destroy_dependents!)
+  end
```
Comment (awisema) [inline app/models/grda_warehouse/data_source.rb]: Should this also destroy the orgs?

File: app/models/grda_warehouse/hud/organization.rb [hunk 2]
```diff
@@ -266,5 +266,9 @@ def self.options_for_select user:
         options
       end
     end
+
+    def destroy_dependents!
+      projects.map(&:destroy_dependents!)
+    end
```
Comment (awisema) [inline app/models/grda_warehouse/hud/organization.rb]: Should this also destroy the projects?

File: app/views/data_sources/index.haml [hunk 3]
```diff
@@ -52,6 +52,5 @@
               %td
                 = checkmark(data_source.authoritative)
               %td
-                - unless data_source.has_data?
-                  = link_to data_source_path(data_source), method: :delete, data: {confirm: "Are you sure you want to delete the #{data_source.name} Data Source?"} do
-                    %span.icon-cross
\ No newline at end of file
+                = link_to data_source_path(data_source), method: :delete, data: {confirm: "Deleting the #{data_source.name} Data Source will delete ALL associated data. Do you want to proceed?"}, class: 'btn btn-sm btn-danger btn-icon-only' do
```
Comment (awisema) [inline app/views/data_sources/index.haml]: Should this be guarded by a permission?

File: app/models/grda_warehouse/data_source.rb (hunk 1 above)
Comment (eanders) [reply app/models/grda_warehouse/data_source.rb]: Yes

File: app/models/grda_warehouse/hud/organization.rb (hunk 2 above)
Comment (eanders) [reply app/models/grda_warehouse/hud/organization.rb]: yes.

File: app/views/data_sources/index.haml (hunk 3 above)
Comment (eanders) [reply app/views/data_sources/index.haml]: Yes.

**Pull request merged**